Services/JWTManager.php

```diff
@@ -88,6 +88,39 @@ public function create(UserInterface $user)
         return $jwtString;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function createFromPayload(UserInterface $user, array $payload)
+    {
+        $payload = array_merge(['roles' => $user->getRoles()], $payload);
+        $this->addUserIdentityToPayload($user, $payload);
```

Comment:
Can you reuse the duplicated logic below by moving it to a private method called from both `create()` and `createFromPayload()`?